crates/chain/src/tx_graph.rs

```diff
@@ -624,14 +634,36 @@ impl<A: Clone + Ord> TxGraph<A> {
     /// The [`ChangeSet`] returned will be empty if graph already knows that `txid` exists in
     /// `anchor`.
     pub fn insert_anchor(&mut self, txid: Txid, anchor: A) -> ChangeSet<A> {
+        let mut old_top_h = None;
+        let mut new_top_h = anchor.confirmation_height_upper_bound();
+
+        let is_changed = match self.anchors.entry(txid) {
+            hash_map::Entry::Occupied(mut e) => {
+                fn top_anchor_height<A: Anchor>(anchors: &BTreeSet<A>) -> Option<u32> {
+                    anchors
+                        .iter()
+                        .last()
+                        .map(Anchor::confirmation_height_upper_bound)
+                }
+                old_top_h = top_anchor_height(e.get());
+                let is_changed = e.get_mut().insert(anchor.clone());
+                new_top_h = top_anchor_height(e.get()).expect("must have anchor");
```

Comment:
🔧 It is a bit hard to understand that what this does is to determine whether a newly inserted anchor has higher height than any of previously existing anchors. It is especially confusing that later the two values are compared as `old_top_h != Some(new_top_h)`.

A comment may help. Or perhaps to find `new_top_h`, compare `old_top_h` with `anchor` directly instead of finding BTreeSet's max value again: either just-inserted `anchor` will be higher or `old_top_h`.

Reply:
Good catch. I tried reorganizing this logic in 87528963805da281c2f674eba63e22a075613117. Let me know what you think.